silo/tasks.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import, unicode_literals
+from celery import shared_task
+from celery.result import AsyncResult
+from tola.celery import app
+
+from silo.custom_csv_dict_reader import CustomDictReader
+from tola.util import saveDataToSilo
+from .models import Silo, Read
+
+import logging
+
+logger = logging.getLogger("tola")
+
+
+@shared_task(bind=True, retry=False)
+def process_silo(self, silo_id, read_id):
+    silo = Silo.objects.get(id=silo_id)
+    read_obj = Read.objects.get(pk=read_id)
+
+    reader = CustomDictReader(read_obj.file_data)
+    saveDataToSilo(silo, reader, read_obj)
+
+    # Todo add notification when done
+    read_obj.task_id = None
+    read_obj.save()
+
+    return True
+
+
+@app.task()
+def process_silo_error(uuid, read_id):
+    result = AsyncResult(uuid)
+    exc = result.get(propagate=False)
+
+    logger.error(exc)
+
+    read_obj = Read.objects.get(pk=read_id)
+    read_obj.task_id = "FAILED"
```

Comment:
I didn't see that you are mixing `task_id` and status in the same field. I don't like it, even it's more compact and you save one field. I think there should be a `task_status` enum field with different states: `CREATED` or `PENDING`, `IN_PROGRESS` and `FINISHED`. This way if something happens and it does not finish, we know at which state it broke, so it makes it easier to debug.

Reply:
Yeah, that's probably better. But i will include `FAILED` in the enum since we know when a task fails.